atomic_reactor/inner.py

```diff
@@ -252,6 +254,52 @@ def all_registries(self):
         return self.docker_registries + self.pulp_registries
 
 
+class FSWatcher(threading.Thread):
+
+    def __init__(self, *args, **kwargs):
+        super(FSWatcher, self).__init__(*args, **kwargs)
+        self.daemon = True  # exits whenever the process exits
+        self._lock = threading.RLock()
+        self._data = {}
+
+    def run(self):
+        while True:
```

Comment:
Ah, I hadn't thought about dangling threads in testing (overhead would still be minimal, but it does feel untidy). The API doesn't provide an obvious way to outright kill a thread, but I guess it would be easy enough to have it exit.